src/main/java/reposense/configparser/CliArgumentsParser.java

```diff
@@ -0,0 +1,101 @@
+package reposense.configparser;
+
+import java.util.Date;
+import java.util.HashMap;
+
+import reposense.exceptions.ParseException;
+
+/**
+ * Verifies and parses user-supplied CLI arguments into an InputParameter Object.
+ */
+public class CliArgumentsParser extends Parser<InputParameter, String[]> {
+
+    private static final String ARGUMENT_PREFIX_DASH = "-";
+
+    private static final String CONFIG_FILE_ARG = "config";
+    private static final String TARGET_FILE_ARG = "output";
+    private static final String SINCE_DATE_ARG = "since";
+    private static final String UNTIL_DATE_ARG = "until";
+
+    private static final String PARSE_EXCEPTION_MESSAGE_INVALID_INPUTS = "Failed to parse inputs arguments";
+    private static final String PARSE_EXCEPTION_MESSAGE_NO_CSV_FILE = "You need to specify a config CSV file!";
+
+    private final HashMap<String, String> argumentMap;
+    private final DateParser dateParser;
+    private final InputParameter argument;
+
+    public CliArgumentsParser() {
```

Comment:
Is there a need to make a constructor it since it only runs for single instance?